src/Microsoft.Identity.Web.TokenAcquisition/ServiceCollectionExtensions.cs

```diff
@@ -160,10 +163,78 @@ public static IServiceCollection AddTokenAcquisition(
                 services.AddScoped<IAuthorizationHeaderProvider, DefaultAuthorizationHeaderProvider>();
             }
 
+            if (enableTokenBinding)
+            {
+                EnableTokenBinding(services);
+            }
+
             services.TryAddSingleton<IMergedOptionsStore, MergedOptionsStore>();
             return services;
         }
 
+        /// <summary>
+        /// Enables token binding by registering required for it services in the service collection.
+        /// </summary>
+        /// <remarks>
+        /// This extension should be called when token acquisition is already configured.
+        /// </remarks>
+        /// <param name="services">The service collection to which the token binding support will be added. Must not be null.</param>
+        /// <returns>The same IServiceCollection instance, allowing for method chaining.</returns>
+        public static IServiceCollection EnableTokenBinding(this IServiceCollection services)
+        {
+            ServiceDescriptor? msalHttpClientFactory = services.FirstOrDefault(s => s.ServiceType == typeof(IMsalHttpClientFactory));
+            if (msalHttpClientFactory != null)
+            {
+                services.Remove(msalHttpClientFactory);
+            }
+
+            // Registering HTTP client factory which allows to create HTTP client with client certificate
+            services.AddSingleton<IMsalHttpClientFactory>(sp =>
```

Comment:
This is fine, but if we can't get it to work at all, it's acceptable to not allow IHttpClient injection in the TokenAcquisition flow and even to disablow usage of DownstreamAPI + POP.